engine/table/src/main/java/io/deephaven/engine/table/impl/DeferredViewTable.java

```diff
@@ -89,51 +87,49 @@ public Table where(Filter filter) {
     }
 
     private Table getResultTableWithWhere(WhereFilter... whereFilters) {
-        if (getCoalesced() != null) {
-            return coalesce().where(Filter.and(whereFilters));
+        {
+            final Table coalesced = getCoalesced();
+            if (coalesced != null) {
+                return coalesced.where(Filter.and(whereFilters));
+            }
         }
 
-        final WhereFilter[] allFilters = Stream.concat(Arrays.stream(deferredFilters), Arrays.stream(whereFilters))
-                .map(WhereFilter::copy).toArray(WhereFilter[]::new);
+        final WhereFilter[] allFilters = Stream.concat(
+                Arrays.stream(deferredFilters).map(WhereFilter::copy),
+                Arrays.stream(whereFilters))
+                .toArray(WhereFilter[]::new);
 
-        TableReference.TableAndRemainingFilters tableAndRemainingFilters;
         if (allFilters.length == 0) {
-            tableAndRemainingFilters = tableReference.getWithWhere();
-            Table result = tableAndRemainingFilters.table;
+            Table result = tableReference.get();
             result = applyDeferredViews(result);
-            result = result.where(Filter.and(tableAndRemainingFilters.remainingFilters));
             copyAttributes((BaseTable<?>) result, CopyAttributeOperation.Coalesce);
             setCoalesced(result);
             return result;
         }
 
-        PreAndPostFilters preAndPostFilters = applyFilterRenamings(allFilters);
-        tableAndRemainingFilters = tableReference.getWithWhere(preAndPostFilters.preViewFilters);
+        final PreAndPostFilters preAndPostFilters = applyFilterRenamings(allFilters);
+        final TableReference.TableAndRemainingFilters tableAndRemainingFilters =
+                tableReference.getWithWhere(preAndPostFilters.preViewFilters);
 
         Table localResult = tableAndRemainingFilters.table;
-        if (localResult instanceof DeferredViewTable) {
-            localResult = ((DeferredViewTable) localResult)
-                    .getResultTableWithWhere(tableAndRemainingFilters.remainingFilters);
-        } else {
-            localResult =
-                    localResult.where(Filter.and(WhereFilter.copyFrom(tableAndRemainingFilters.remainingFilters)));
+        if (tableAndRemainingFilters.remainingFilters.length != 0) {
+            localResult = localResult.where(Filter.and(tableAndRemainingFilters.remainingFilters));
```

Comment:
does Filter.and do something useful here?

Reply:
Yes. The API is written to accept one filter, instead of `List`.